sdk/appconfiguration/app-configuration/test/public/tracing.spec.ts

```diff
@@ -45,11 +62,82 @@ describe("supports tracing", () => {
         "AppConfigurationClient.deleteConfigurationSetting",
       ],
     );
-    try {
-      await client.setReadOnly({ key: key }, false);
-      await client.deleteConfigurationSetting({ key: key });
-    } catch (e: any) {
-      /** empty because key is already deleted */
-    }
+  });
+
+  it("can trace through the various options", async function () {
+    console.log("expected to pass with all spans");
+    const key = recorder.variable(
+      "noLabelTests",
+      `noLabelTests${Math.floor(Math.random() * 1000)}`,
+    );
+    await assert.supportsTracing(
+      async (options) => {
+        const promises: Promise<any>[] = [
+          fixedClient.addConfigurationSetting({ key }, options),
+          fixedClient.getConfigurationSetting({ key }, options),
+          fixedClient.setConfigurationSetting({ key, value: "new-value" }, options),
+          fixedClient.setReadOnly({ key }, true, options),
+          fixedClient.deleteConfigurationSetting({ key }, options),
+        ];
+        // We don't care about errors, only that we created (and closed) the appropriate spans.
+        await Promise.all(promises.map((p) => p.catch(() => undefined)));
+      },
+      [
+        "AppConfigurationClient.addConfigurationSetting",
+        "AppConfigurationClient.getConfigurationSetting",
+        "AppConfigurationClient.setConfigurationSetting",
+        "AppConfigurationClient.setReadOnly",
+        "AppConfigurationClient.deleteConfigurationSetting",
+      ],
+    );
+  });
+
+  it("can trace through the various options", async function () {
```

Comment:
This is the other option, which we can centralize of course and clean up, but the TL;DR; is

- mock in some way that you can check the span names 
- check that withSpan was called N times
- check that the span names match

I like the other option better

Reply:
To me the mock approach is more of a true unit test, but if we're really trying to validate scenario correctness, the extra plumbing of passing the instrumenter seems fairly low overhead and feels good from a dependency injection POV.

Reply:
Yeah the more I look at it the more I am on the fence 😄 I like that we don't have to add anything to the public API to support these tests and I think the test coverage is sufficient - core-rest-pipeline which outputs the most important spans (the HTTP spans) doesn't use this helper today so that remains unchanged.

I think out of the two, what will help me is to sync up with @mpodwysocki and get a better understanding on which approach will play nicer with vitest and ESM. That will likely help sway one way or another... 

If we decided to go this approach though, I can own updating all the packages and tests  - it'll be tedious but not too hard...

Reply:
I'm with @xirzec that we should test for correctness and passing through with dependency injection seems a bit nicer than replacing globals.

Reply:
SGTM, I created this issue to track that work https://github.com/Azure/azure-sdk-for-js/issues/29736 as a longer term fix